vector/src/main/java/im/vector/app/features/matrixto/MatrixToBottomSheetViewModel.kt

```diff
@@ -125,42 +124,23 @@ class MatrixToBottomSheetViewModel @AssistedInject constructor(
     }
 
     private fun handleStartChatting(action: MatrixToAction.StartChattingWithUser) {
-        val mxId = action.matrixItem.id
-        val existing = session.getExistingDirectRoomWithUser(mxId)
-        if (existing != null) {
-            // navigate to this room
-            _viewEvents.post(MatrixToViewEvents.NavigateToRoom(existing))
-        } else {
+        viewModelScope.launch {
             setState {
```

Comment:
Set loading state before launching in coroutine scope

Reply:
why?